server/controllers/index.js

```diff
@@ -155,11 +194,18 @@ router.post('/api/voteCard', auth, (req, res) => {
 /* Check if in dev mode, and enable end game request */
 if (process.env.NODE_ENV === 'testing') {
     router.post('/api/reset-server', (req, res) => {
-        gameLogic.clearGameState();
-        closeSocket();
-        req.session.destroy();
-        currentUsers = [];
-        res.sendStatus(200);
+        try {
+            if (req.session.roomId) {
+                getGameStateById(req.session.roomId).clearGameState();
+            }
+            closeSocket();
+            req.session.destroy();
+            currentUsers = [];
+            res.sendStatus(200);
+        } catch (err) {
+            console.log(err);
+            res.status(400).json({ message: err.message});
+        }
```

Comment:
Concerned that here if there was an error the socket would never be closed, destroying resources should ideally take place in a "finally" block so that it happens regardless

Reply:
But do we want to close the socket if the server cannot end the game successfully?

Reply:
This only gets called in tests for resetting the server so we probably do wanna reset everything no matter what

Reply:
okay, I'll put it in a finally block